apps/rush-lib/src/scripts/install-run.ts

```diff
@@ -445,13 +445,12 @@ export function installAndRun(
 
   const binPath: string = _getBinPath(packageInstallFolder, packageBinName);
   const binFolderPath: string = path.resolve(packageInstallFolder, NODE_MODULES_FOLDER_NAME, '.bin');
+  process.env.PATH = [binFolderPath, process.env.PATH].join(path.delimiter);
```

Comment:
Wouldn't this change the PATH environment variable for this script (and all subprocesses)?

Reply:
That was sort of the point :-) But instead of trying to accommodate OS-specific behavior, this relies on Node doing the right thing.

Is there a concern about leaking the updated path?

Reply:
If there is such a concern, we could capture and restore the current PATH + check if we have to update it, something like

````typescript
  const path:string = process.env.PATH;
  
  if (path.indexOf(binFolderPath) === -1) {
    process.env.PATH = [binFolderPath, process.env.PATH].join(path.delimiter);
  }

  const result: childProcess.SpawnSyncReturns<Buffer> = childProcess.spawnSync(...);

  process.env.PATH = path;
````

However, as my understanding is that installAndRun() is only invoked once at the tail of the "install-run" script, I opted for the simplest/smallest change that solves the problem we're facing.

Reply:
> this relies on Node doing the right thing.

I am not sure how your PR does that though.

Reply:
@octogonz Do you want to take a look at this PR?

Reply:
Consider these two experiments:
```
process.env['XXX'] = 123;   
process.env['xxx'] = 321;  
console.log(process.env);
// prints all the variables, but only one for `xxx`
```

```
process.env = { 'XXX': 123, 'xxx': 321 };
console.log(process.env);
// prints two values: { XXX: 123, xxx: 321 }
```

```
process.env = { };
process.env['XXX'] = 123;   
process.env['xxx'] = 321;  
console.log(process.env);
// prints { xxx: 321, XXX: 123 }
```

In other words, the original `process.env` proxy merges different-cased IDs.  Whereas ordinary JavaScript objects do not, and assigning anything to `process.env` overwrites the proxy with a plain old JavaScript value.

This is potentially a bug that affects all our `process.env` manipulations. Let's think about a comprehensive solution to that.

Reply:
The Node.js normalization happens in [node_env_var.cc](https://github.com/nodejs/node/blob/7a447bcd54f5e153d14da377963541349d7bda09/src/node_env_var.cc#L299) native code, which eventually goes to [RealEnvStore::Set](https://github.com/nodejs/node/blob/6726246dbb83e3251f080fc4729154d492f7e340/src/node_env_var.cc#L116) and on Windows that ultimately invokes [SetEnvironmentVariableW](https://github.com/nodejs/node/blob/bb173f931ae85b464e901806f15dcc219dec6b73/src/node.cc#L2714) which is case-insensitive.  This native proxy is bound to `process.env` when the `process` module is initialized. 

Ideally we should rely on that logic, so we don't have to invent our own design.  

However, it seems that if any script anywhere does `process.env = {};`, then the case-insensitive proxy is replaced with a plain old JavaScript object, which is case-sensitive.  This effect is global for all Node.js modules in the process.  If I then use `child_process.execSync()` to spawn a child process, Windows 10 somehow creates an environment with mixed case.  Thus it seems that some Windows APIs enforce a case-insensitive environment (`SetEnvironmentVariableW`?) but others do not (`CreateProcess`?).

Reply:
Okay here is one more experiment:
```js
process.env = {};
process.env['__TEST'] = 123;
process.env['__test'] = 321;

child_process.execSync('cmd.exe /c set', { shell: true, stdio: 'inherit' });
child_process.execSync('cmd.exe /c echo Expanded __TEST to be %__TEST%', { shell: true, stdio: 'inherit' });
```
It prints:
```
__test=321
__TEST=123
Expanded __TEST to be 321
```

In other words, when the environment is mixed case, the `GetEnvironmentVariableW` returns the first case-insensitive match that it finds, ignoring the other one.  From my experiments, the order of precedence is something arbitrary and not alphabetical.

This convinces me that the Windows API is too screwy to be useful, even if we could find a reliably way to access the `EnvSetter`/`EnvGetter` native proxy. 

Instead I'm thinking that `node-core-library` should provide an API  (`EnvironmentTable`?) that will automatically deduplicate different-cased keys on Windows.  I'm sure mixed case is required for some special cases like Bash interoperability, but to the extent that Rush will be invoking Windows shell commands, mixed case seems simply too dangerous to tolerate. We should normalize that away.

I'll create a separate PR to propose that general solution.  It won't be useful for the **install-run.ts** however, because it is a self-contained script. @crhaglun's solution here is probably. Lemme look at it one more time, and then we'll get this PR merged to unblock people.

(What an interesting rabbit hole!)

Reply:
> This convinces me that the Windows API is too screwy to be useful

Well, no argument from me there, thank you very much for that extra investigation to confirm more precisely what the problem is.

Reply:
Thanks @octogonz!